src/purpleLayer.js

```diff
@@ -0,0 +1,120 @@
+require('jquery') ; 
+require('leaflet') ; 
+require('heatmap.js') ;
+require('leaflet-heatmap') ;
+
+L.LayerGroup.PurpleLayer = L.LayerGroup.extend(
+
+    {
+        
+        options: {
+            "radius": 2,
+            "maxOpacity": 1 , 
+            "scaleRadius": true, 
+            "useLocalExtrema": true,
+             latField: 'lat',
+             lngField: 'lng',
+             valueField: 'count' ,
+             blur: .75    
+        },
+        
+        initialize: function (options) {
+            options = options || {};
+            L.Util.setOptions(this, options); 
+            this._layers = {};  
+            this._purpleLayerArray = [] ;
+            this.heatmapLayer ;
+        },
+        
+        onAdd: function (map) {
+             this._map = map;
+             this.heatmapLayer = new HeatmapOverlay(this.options) ;
+           
+             this.requestData();
+
+
+        },
+        
+        onRemove: function (map) {
+            this._map.removeLayer(this.heatmapLayer) ;
+            this.clearLayers();
+            this._layers = {};
+        },
+        
+        requestData: function () {
+           var self = this;
+                (function() {
+                    var script = document.createElement("SCRIPT");
+                    script.src = 'https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js';
+                    script.type = 'text/javascript';
+                    
+                    script.onload = function() {
+                        var $ = window.jQuery;
+                        var PurpleLayer_url = "https://www.purpleair.com/json?fetchData=true&minimize=true&sensorsActive2=10080&orderby=L";
+                        $.getJSON(PurpleLayer_url , function(data){
+                        	 self.parseData(data) ;    
+            		    });
+                    };
+                    document.getElementsByTagName("head")[0].appendChild(script);
+                })(); 
+            
+            
+        },
+       
+        getMarker: function (data) {
+              var lat = data.Lat ;
+              var lng = data.Lon;
+              var value = parseFloat(data.PM2_5Value) ;  //PM2.5 VALUE in microgram per metre cube
+
+              var purpleLayer_object = new Object() ; 
+              purpleLayer_object.lat = lat ;
+              purpleLayer_object.lng = lng ;
+
+              var aqi ;
+              if(value>=0 && value<=12.0)
```

Comment:
@jywarren ...used the formula given on Wikipedia page !
The AQI value calculated matches with the one calculated here : https://www.purpleair.com/map?&zoom=4&lat=39.96179960675426&lng=-74.31505939400137&clustersize=45&orderby=L&latr=46.230881869445966&lngr=126.56250000000006   🎉 😄  .